gk/main.c

```diff
@@ -2219,6 +1512,71 @@ process_cmds_from_mailbox(
 	mb_free_entry_bulk(&instance->mb, (void * const *)gk_cmds, num_cmd);
 }
 
+static void
+populate_front_tasks(struct gk_co_work *work,
+	uint16_t port_front, uint16_t rx_queue_front)
+{
+	uint16_t front_max_pkt_burst = work->gk_conf->front_max_pkt_burst;
+	struct rte_mbuf *rx_bufs[front_max_pkt_burst];
+	/* Load a set of packets from the front NIC. */
+	uint16_t num_rx = rte_eth_rx_burst(port_front, rx_queue_front, rx_bufs,
+		front_max_pkt_burst);
+	struct gk_measurement_metrics *stats;
+	bool has_rss;
+	int i;
+
+	if (unlikely(num_rx == 0))
+		return;
+
+	stats = &work->instance->traffic_stats;
+	stats->tot_pkts_num += num_rx;
+
+	has_rss = work->gk_conf->net->front.rss;
+	for (i = 0; i < num_rx; i++) {
+		struct gk_co_task *task = &work->all_tasks[work->task_num++];
+		struct rte_mbuf *pkt = rx_bufs[i];
+
+		stats->tot_pkts_size += rte_pktmbuf_pkt_len(pkt);
+
+		if (likely(has_rss)) {
+			task->task_hash = pkt->hash.rss;
```

Comment:
We must rewrite `pkt->hash.rss` with `pkt->hash.rss >> rss_bits`, where `rss_bits` is $\lceil log_2(n) \rceil$ and $n$ the number of RSS queues, which is equal to the number of instances of the GK (or GT) block. This is necessary because the $n$ least significative bits of `pkt->hash.rss` are constant for a given GK instance.